proposals/interface-types/working-notes/Types.md

```diff
@@ -0,0 +1,99 @@
+# Interface value types
+
+This proposal defines new interface values, which have corresponding interface value types. Interface value types are a superset of the [core value types](https://webassembly.github.io/spec/core/syntax/types.html#value-types)
+
+Some interface value types are compound by being constructed using one or more interface value types. The abstract syntax is designed to prevent recursive types by not allowing a compound type definition to refer to itself or other definitions. This is so that the set of control flow needed to generate all interface values is simple enough to allow for optimized lazy semantics.
+
+The interface value type syntax is summarized below. Following after that is a section for each type that gives the full syntax, possible values, validation rules, and subtyping relationships.
+
+```
+#interface-valtype ::=
+  | #valtype
+  | bool
+  | s8
+  | s16
+  | s32
+  | s64
+  | u8
+  | u16
+  | u32
+  | u64
+  | string
+  | record ..
+  | array ..
+```
+
+## `bool` 
+
+A boolean value. The set of `{true, false}`.
```

Comment:
would really prefer to have this fall out of the variant type story; with bool being a macro of a type not a first class one.

Reply:
That makes sense to me. I'm going to cut out `bool` for now until we have `variant`.